packages/cli/src/oclif/commands/validate.js

```diff
@@ -0,0 +1,108 @@
+const colors = require('colors/safe');
+const { flags } = require('@oclif/command');
+
+const BaseCommand = require('../ZapierBaseCommand');
+const utils = require('../../utils');
+const { buildFlags } = require('../buildFlags');
+const { flattenCheckResult, formatStyles } = require('../../utils/display');
+
+class ValidateCommand extends BaseCommand {
+  async perform() {
+    this.log('Validating project locally.');
+
+    const errors = await utils.localAppCommand({ command: 'validate' });
+
+    const newErrors = errors.map(error => {
+      error = Object.assign({}, error);
+      error.property = error.property.replace(/^instance/, 'App');
+      error.docLinks = (error.docLinks || []).join('\n');
+      return error;
+    });
+    this.logTable({
+      rows: newErrors,
+      headers: [
+        ['Property', 'property'],
+        ['Message', 'message'],
+        ['Links', 'docLinks']
+      ],
+      emptyMessage: colors.grey(
+        'No structural errors found during validation routine.'
+      )
+    });
+
+    if (newErrors.length) {
+      this.log(
+        'Your app is structurally invalid. Address concerns and run this command again.'
+      );
+      process.exitCode = 1;
+    } else {
+      this.log('This project is structurally sound!');
+    }
+
+    let checkResult = [];
+    if (0 && (this.flags['no-server'] || process.exitCode === 1)) {
+      if (process.exitCode === 1) {
+        this.log(
```

Comment:
is there a reason not to use `this.error(message)` instead and skip the rest of the function?

Reply:
This was how it is in the original command. I also think for `zapier validate` this is a normal case instead of an error. `zapier validate`'s job is to find violations, so it's normal that the app didn't validate. I believe `this.error` is for real errors. For `zapier validate`, it could be `zapier-platform-core` hasn't been installed yet or the developer runs in a random directory.

Reply:
fair! there's also a built-in `this.warn`. logs with a yellow `>`, iirc. A little more eye-catching, but doesn't throw.